src/Persister/Generic.php

```diff
@@ -47,6 +47,7 @@ public function getName() {
   public function persist(&$entity, $pid, $save = TRUE) {
     $target_field = trim($this->config->get('persistent_identifiers_target_field'));
     if (method_exists($entity, 'hasField') && $entity->hasField($target_field)) {
+      // TODO: Don't add same values
```

Comment:
Something like this maybe?

```php
$existing_values = $entity->get($target_field)->getValue()
foreach ($existing_values as $value) {
    if ($pid != $value) {
        $entity->{$target_field}[] = $pid;
    }
}
```

Reply:
Or maybe better to use a `!in_array($pid, $existing_values)`?

Reply:
an !in_array  is better, because looping through the field, if they have multiple values causes additional appends to the array.